scheduler-interface-spec.md

```diff
@@ -356,67 +362,61 @@ message AllocationAsk {
-  Resource resourceAsk = 2;
-
-  // Tags of Allocation
-  map<string, string> tags = 3;
-
-  // Priority of ask
-  Priority priority = 4;
-
-  // Placement constraint defines how app wanna to be placed in the cluster.
-  // if not set, no placement constraint will be applied.
-  PlacementConstraint placementConstraint = 6;
-
+  // The application ID this allocation ask belongs to
+  string applicationId = 2;
+  // The partition the application belongs to
+  string partitionName = 3;
+  // The amount of resources per ask
+  Resource resourceAsk = 4;
   // Maximum number of allocations
-  int32 maxAllocations = 7;
-
-  // Who owns the allocation
-  UserGroupInformation ugi = 8;
-
-  // Place this allocation to specified queue.
-  // If queue not specified, scheduler will place the allocation to a queue
-  // according to policy
-  // When queue is specified and allocation cannot be allocated in asked queue
-  // AllocationAsk will be rejected.
-  string queueName = 9;
-
+  int32 maxAllocations = 5;
+  // Priority of ask
+  Priority priority = 6;
   // Execution timeout: How long this allocation will be terminated (by scheduler)
   // once allocated by scheduler, 0 or negative value means never expire.
-  int64 executionTimeoutMilliSeconds = 10;
-  
-  // Which application this allocation ask belongs to, if it is not specified. The allocation
-  // won't belong to any application.
-  string applicationId = 11;
-  
-  // Which partition requested by this ask
-  // One ask can only request one node partition
-  string PartitionName = 12;
-  
+  int64 executionTimeoutMilliSeconds = 7;
+  // Placement constraint defines how this allocation should be placed in the cluster.
+  // if not set, no placement constraint will be applied.
+  PlacementConstraint placementConstraint = 8;
+
   // TODO.
   // wangda: even if I don't like it, it seems we have to support old-school style
   // YARN resource requests, otherwise it gonna be very hard to support legacy apps
   // running on YARN.
   // Maybe a simple approach is to add a flag to indicate it is an old-school resource
   // request from YARN.
-  // Check: YARN code: placement.LocalityAppPlacementAllocator for more details. 
+  // Check: YARN code: placement.LocalityAppPlacementAllocator for more details.
 }
 
 message UserGroupInformation {
+  // the user name
   string user = 1;
+  // the list of groups of the user, can be empty
   repeated string groups = 2;
 }
 
 message AddApplicationRequest {
+  // The ID of the application, must be unique
   string applicationId = 1;
+  // The queue this application is requesting. The scheduler will place the allocation to a
+  // queue according to policy, taking into account the requested queue as per the policy.
   string queueName = 2;
+  // The partition the application belongs to
   string partitionName = 3;
+  // The user group information of the application owner
+  UserGroupInformation ugi = 4;
+  // A set of tags for the application
```

Comment:
We need to add some more clear comment for this, letting people know this is a map where to store app-level generic info.